packages/react/src/components/customSort/index.tsx

```diff
@@ -0,0 +1,180 @@
+import {
+  Context,
+  getSheetIndex,
+  indexToColumnChar,
+  locale,
+  sortSelection,
+} from "@fortune-sheet/core";
+import React, {
+  ChangeEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import _ from "lodash";
+import WorkbookContext from "../../context";
+import "./index.css";
+import { useDialog } from "../../hooks/useDialog";
+
+type RadioChangeEvent = React.ChangeEvent<HTMLInputElement>;
+
+const CustomSort: React.FC<{}> = () => {
+  const [rangeColChar, setRangeColChar] = useState<String[]>([]);
+  const [ascOrDesc, setAscOrDesc] = useState(true);
+  const { context, setContext } = useContext(WorkbookContext);
+  const [selectedValue, setSelectedValue] = useState<string>("0");
+  const [isTitleChange, setIstitleChange] = useState(false);
+  const { sort } = locale(context);
+  const { hideDialog } = useDialog();
+
+  const col_start = context.luckysheet_select_save![0].column[0];
+  const col_end = context.luckysheet_select_save![0].column[1];
+  const row_start = context.luckysheet_select_save![0].row[0];
+  const row_end = context.luckysheet_select_save![0].row[1];
+
+  const sheetIndex = getSheetIndex(context, context.currentSheetId) as number;
+
+  const handleSelectChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSelectedValue(event.target.value);
+  };
+
+  // 改变排序方式
+  const handleRadioChange = useCallback((e: RadioChangeEvent) => {
+    const sortValue = e.target.value;
+    setAscOrDesc(() => {
+      if (sortValue === "asc") {
+        return true;
+      }
+      return false;
+    });
+  }, []);
+
+  const handleTitleChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const value = e.target.checked;
+      setIstitleChange(value);
+    },
+    []
+  );
+
+  // 获取排序列
+  useEffect(() => {
+    setRangeColChar([]);
+    if (isTitleChange) {
+      for (let i = col_start; i <= col_end; i += 1) {
+        // 判断列首是否为空
+        if (
+          !_.isNil(context.luckysheetfile[sheetIndex].data![row_start][i]?.v)
+        ) {
+          const colHeaderValue = context.luckysheetfile[sheetIndex].data![
+            row_start
+          ][i]!.v as string;
+          setRangeColChar((prevArray) => [...prevArray, colHeaderValue]);
+        } else {
+          const ColumnChar = indexToColumnChar(i);
+          setRangeColChar((prevArray) => [...prevArray, `列${ColumnChar}`]);
+        }
+      }
+    } else {
+      for (let i = col_start; i <= col_end; i += 1) {
+        const ColumnChar = indexToColumnChar(i);
+        setRangeColChar((prevArray) => [...prevArray, ColumnChar]);
+      }
+    }
+  }, [
+    col_end,
+    col_start,
+    context.luckysheetfile,
+    isTitleChange,
+    row_start,
+    sheetIndex,
+  ]);
```

Comment:
```suggestion
  useEffect(() => {
    const list: string[] = [];
    if (isTitleChange) {
      for (let i = col_start; i <= col_end; i += 1) {
        // 判断列首是否为空
        const cell = context.luckysheetfile[sheetIndex].data?.[row_start]?.[i];
        const colHeaderValue = cell?.m || cell?.v;
        if (colHeaderValue) {
          list.push(colHeaderValue as string);
        } else {
          const ColumnChar = indexToColumnChar(i);
          list.push(`${sort.columnOperation} ${ColumnChar}`);
        }
      }
    } else {
      for (let i = col_start; i <= col_end; i += 1) {
        const ColumnChar = indexToColumnChar(i);
        list.push(ColumnChar);
      }
    }
    setRangeColChar(list);
  }, [
    col_end,
    col_start,
    context.luckysheetfile,
    isTitleChange,
    row_start,
    sheetIndex,
    sort.columnOperation,
  ]);
```

Calling state setter in a for-loop is bad practice.